codes/ruby/chapter_backtracking/preorder_traversal_i_compact.rb

```diff
@@ -0,0 +1,33 @@
+=begin
+File: preorder_traversal_i_compact.rb
+Created Time: 2024-04-07
+Author: hoangtien2k3 (hoangtien2k3qx1@gmail.com)
+=end
+
+require_relative 'utils'
+
+# 前序遍历：例题一
+def pre_order(root)
+  return if root.nil?
+
+  if root.val == 7
+    # 记录解
+    $res << root
+  end
+
+  pre_order(root.left)
+  pre_order(root.right)
+end
+
+root = TreeNode.list_to_tree([1, 7, 3, 4, 5, 6, 7])
+puts "\n初始化二叉树"
+PrintUtil.print_tree(root)
```

Comment:
There is no `PrintUtil` module for the consistency with other programming languages' code. Just `print_tree` method is enough. Please check the `utils/*.rb` to see what kind of helper methods it is providing.